bitfield.go

```diff
@@ -388,11 +388,7 @@ func (bf *BitField) AllMap(max uint64) (map[uint64]bool, error) {
 	return out, nil
 }
 
-func (bf *BitField) MarshalCBOR(w io.Writer) error {
-	if bf == nil {
-		_, err := w.Write(cbg.CborNull)
-		return err
-	}
+func (bf BitField) MarshalCBOR(w io.Writer) error {
```

Comment:
Should this remain with a pointer receiver to parallel UnmarshalCBOR? (but fail if nil)

You probably know better than I, but I've run into unspecified trouble before with these de/serialization methods on value types.